DylibBinder/DBFunc.cs

```diff
@@ -0,0 +1,141 @@
+			// TODO not sure how to handle instances with protocols yet
+			if (isProtocol)
+				HasInstance = "False";
+
+			ReturnType = SwiftTypeToString.MapSwiftTypeToString (tlf.Signature.ReturnType, tlf.Module.Name);
+			ParameterLists = new DBParameterLists (tlf.Signature, HasInstance);
+			GenericParameters = new DBGenericParameters (tlf.Signature);
+		}
+
+		public DBFunc (DBProperty dbProperty, string propertyType, bool isMutating = false)
+		{
+			if (propertyType == "Getter")
+				Name = $"get_{dbProperty.Name}";
+			else
+				Name = $"set_{dbProperty.Name}";
+			PropertyType = propertyType;
+			IsStatic = dbProperty.IsStatic;
+			IsProperty = "True";
+			HasThrows = "False";
+			IsMutating = isMutating.ToString ();
+			HasInstance = IsStatic == "True" ? "False" : "True";
+			ReturnType = dbProperty.Type;
+			ParameterLists = new DBParameterLists (ReturnType, HasInstance, propertyType);
+			GenericParameters = dbProperty.GenericParameters;
+		}
+
+		public string Name { get; }
+		public string IsStatic { get; }
+		public string IsProperty { get; }
+		public string ReturnType { get; }
+		public string HasThrows { get; }
+		public string IsMutating { get; }
+		public string HasInstance { get; }
+
+		public string PropertyType { get; } = "";
+		public string IsPossiblyIncomplete { get; } = "False";
+		public string OperatorKind { get; } = "None";
+		public string Accessibility { get; } = "Public";
+		public string IsFinal { get; } = "False";
+		public string IsDeprecated { get; } = "False";
+		public string IsUnavailable { get; } = "False";
+		public string IsOptional { get; } = "False";
+		public string IsRequired { get; } = "False";
+		public string IsConvenienceInit { get; } = "False";
+		public string ObjcSelector { get; } = "";
+
+		public DBParameterLists ParameterLists { get; }
+		public DBGenericParameters GenericParameters { get; }
+		public List<DBAssociatedType> AssociatedTypes { get; } = new List<DBAssociatedType> ();
+
+		string GetStaticStatus (SwiftBaseFunctionType signature)
+		{
+			if (signature is SwiftStaticFunctionType)
+				return "True";
+			else if (signature is SwiftStaticFunctionThunkType)
+				return "True";
+			else if (signature is SwiftPropertyType propType)
+				return propType.IsStatic.ToString ();
+			else if (signature is SwiftPropertyThunkType propThunkType)
+				return propThunkType.IsStatic.ToString ();
+			else if (signature is SwiftUncurriedFunctionThunkType)
+				return "False";
+			else if (signature is SwiftUncurriedFunctionType)
+				return "False";
+
+			return "False";
+		}
+	}
+
+	public class DBFuncs {
+		public DBFuncs (ClassContents classContents)
+		{
+			var functions = classContents.Methods.Values.ToList ();
+			functions.AddRange (classContents.StaticFunctions.Values.ToList ());
+			functions.AddRange (classContents.Constructors.Values.ToList ());
+			functions.Sort ((type1, type2) => String.CompareOrdinal (type1.Name.Name, type2.Name.Name));
```

Comment:
How may times are we going to write 'Sort ((type1, type2) => String.CompareOrdinal (type1.Name.Name, type2.Name.Name))' in this project?

Reply:
So about this sort, I call it 3 different times but the List type and the "path" of the properties to the string to sort by are all different. Can I still make this modular?